sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -26,28 +27,29 @@
  */
 public class HttpLoggingPolicy implements HttpPipelinePolicy {
     private static final ObjectMapper PRETTY_PRINTER = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
-    private final HttpLogDetailLevel detailLevel;
+    private final HttpLogOptions httpLogOptions;
     private final boolean prettyPrintJSON;
     private static final int MAX_BODY_LOG_SIZE = 1024 * 16;
+    private static final String REDACTED_PLACEHOLDER = "REDACTED";
 
     /**
-     * Creates an HttpLoggingPolicy with the given log level.
+     * Creates an HttpLoggingPolicy with the given log configurations.
      *
-     * @param detailLevel The HTTP logging detail level.
+     * @param httpLogOptions The HTTP logging configurations.
      */
-    public HttpLoggingPolicy(HttpLogDetailLevel detailLevel) {
-        this(detailLevel, false);
+    public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
+        this(httpLogOptions, false);
     }
 
     /**
-     * Creates an HttpLoggingPolicy with the given log level and pretty printing setting.
+     * Creates an HttpLoggingPolicy with the given log configuration and pretty printing setting.
      *
-     * @param detailLevel The HTTP logging detail level.
-     * @param prettyPrintJSON If true, pretty prints JSON message bodies when logging.
-     *     If the detailLevel does not include body logging, this flag does nothing.
+     * @param httpLogOptions The HTTP logging configuration options.
+     * @param prettyPrintJSON If true, pretty prints JSON message bodies when logging. If the detailLevel does not
+     * include body logging, this flag does nothing.
      */
-    public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON) {
-        this.detailLevel = detailLevel;
+    public HttpLoggingPolicy(HttpLogOptions httpLogOptions, boolean prettyPrintJSON) {
```

Comment:
I'm curious if, outside of unit tests, anyone calls this constructor. If no one does, I would like to see this constructor made package-private for now, because this kind of boolean config should be in the HttpLogOptions object.